### PR TITLE
Ice cream machine tg UI interface conversion

### PR DESCRIPTION
**Pull request by EleSeu**
[CATERING] [UI]

## About the PR 
Converts the Ice Cream machine UI to TGUI. It also makes possible to see what it's inside the beaker loaded into the machine, and makes it clearer that it is a possibility to make custom ice creams.
image

## Why's this needed? 
New UI to make it consistent with the rest of the game and easier to use. 

## Changelog 

```changelog
(u)Elettra
(+)New UI for the Ice Cream machine using TGUI.
```

### CODE REVIEW

File: tgui/packages/tgui/interfaces/IceCreamMachine.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        </Button>
+      </Flex.Item>
+    </Flex>
+  );
+};
+
+export const IceCreamMachine = (props, context) => {
+  return (
+    <Window
+      title="Ice Cream-O-Mat 9900"
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/IceCreamMachine.js:99]:
```suggestion
      title="Ice Cream-O-Mat 6300"
```
i honestly didn't know this was actually a pun name until recently (ice-o-mat is a brand of ice crushing machines)
version will be downgraded to allow for future improvements to the UI

Comment (EleSeu) [reply tgui/packages/tgui/interfaces/IceCreamMachine.js]: Cool, I didn't know that. I changed the name anyway

**Review by Tarmunora — APPROVED** [review]

File: tgui/packages/tgui/interfaces/IceCreamMachine.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  return (
+    <Section title="Standard Flavors" >
+      <Flex wrap>
+        {standardFlavors.map((flavor, flavorIndex) => (
+          <Button
+            key={flavorIndex}
+            align="left"
+            width="130px"
+            m=".1rem"
+            onClick={() => act("make_ice_cream", { flavor: flavor.name })}
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/IceCreamMachine.js:26]: All other dispensers like this have the reagent names capitalised

File: tgui/packages/tgui/interfaces/IceCreamMachine.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+              Make Custom Ice Cream
+            </Button>
+          </Flex.Item>
+          <Flex.Item>
+            <Button
+              mt=".5rem"
+              icon="eject"
+              disabled={!beaker}
+              onClick={() => act("eject_beaker")} >
+              {!beaker ? "Eject Beaker" : "Eject " + beaker.name }
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/IceCreamMachine.js:68]: This should match how the buttons on chem dispensers work, with the container name and an insert option if there is no container

File: tgui/packages/tgui/interfaces/IceCreamMachine.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        <Flex wrap >
+          <Flex.Item>
+            <Button key="beaker"
+              mt=".5rem"
+              mr=".5rem"
+              icon="check" color="green"
+              disabled={!beaker || !beaker.totalVolume}
+              tooltip={beaker && !beaker.totalVolume ? "Beaker Is Empty" : ""}
+              onClick={() => act("make_ice_cream", { flavor: "beaker" })}>
+              Make Custom Ice Cream
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/IceCreamMachine.js:59]: The capitalisation on these two buttons doesn't match

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/IceCreamMachine.js (hunk 2 above)
Comment (ZeWaka) [reply tgui/packages/tgui/interfaces/IceCreamMachine.js]: i don't think this was addressed?

Comment (EleSeu) [reply tgui/packages/tgui/interfaces/IceCreamMachine.js]: Using className="chem-dispenser__dispense-buttons" makes the buttons capitalised

**Pull request merged**